app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class CartsController < ApplicationController
+  def show
+    @cart = Cart.where(user_id: session[:user_id]) if current_user
+  end
+
+  def create
+    if current_user
+      cart = current_user.cart.find_by(item_id: params[:item_id])
+      if cart
+        params[:quantity] += cart.quantity
+        cart.update(cart_params)
+      else
+        current_user.cart.new(cart_params)
+      end
+      if item.save
+        flash[:success] = "Item added to cart!"
+        redirect_to root_path
```

Comment:
We need to readjust these routes from root_path to wherever they actually need to go.